### PR TITLE
Fix setting parameter from station config

### PR DESCRIPTION
**Pull request by Dominik-Vogel**
This PR fixes a bug reported by @ThorvaldLarsen for setting-up a parameter of a `ChannelList` through the station config file.
Consider the following example:
```yaml
instruments:
  mock:
    type: qcodes.tests.instrument_mocks.DummyChannelInstrument
    parameters:
      channels.temperature:
          initial_value: 10
```
Here `channels` is of type `ChannelList`.

### CODE REVIEW

File: qcodes/station.py [hunk 1]
```diff
@@ -52,6 +53,9 @@ def get_config_use_monitor() -> Optional[str]:
     return qcodes.config["station"]["use_monitor"]
 
 
+ChannelOrInstrumentBase = Union[InstrumentBase, ChannelList]
```
Comment (astafan8) [inline qcodes/station.py]: shouldn't this also be a channel? (not only channel lists) some drivers have "submodules" with parameters implemented via subclassing InstrumentChannel.
```suggestion
ChannelOrInstrumentBase = Union[InstrumentBase, ChannelList, InstrumentChannel]
```

File: qcodes/station.py [hunk 2]
```diff
@@ -456,31 +460,47 @@ def load_instrument(self, identifier: str,
...
-        # local function to refactor common code from defining new parameter
-        # and setting existing one
-        def resolve_parameter_identifier(instrument: InstrumentBase,
-                                         identifier: str) -> Parameter:
+        def resolve_instrument_identifier(
+            instrument: ChannelOrInstrumentBase,
+            identifier: str
+        ) -> ChannelOrInstrumentBase:
+            """
+            Get the instrument or channel_list described by a nested string.
```
Comment (astafan8) [inline qcodes/station.py]: shouldn't this be both channel_list and channel?

File: qcodes/station.py (hunk 1 above)
Comment (Dominik-Vogel) [reply qcodes/station.py]: It is :-)
```python
class InstrumentChannel(InstrumentBase):
```

Comment (astafan8) [reply qcodes/station.py]: oh that's super good! stupid me :)

**Comment by Dominik-Vogel** [comment]
This was a premature conversion from draft (I could swear I had the tests passing). I'll notify you once its good to go.

**Comment by Dominik-Vogel** [comment]
Ok, now it's good to go, @astafan8 @jenshnielsen

**Pull request merged**